main/src/com/google/refine/browsing/facets/TextListFacet.java

```diff
@@ -0,0 +1,172 @@
+    * Neither the name of Google Inc. nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,           
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY           
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+package com.google.refine.browsing.facets;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.refine.ProjectManager;
+import com.google.refine.browsing.DecoratedValue;
+import com.google.refine.browsing.FilteredRecords;
+import com.google.refine.browsing.FilteredRows;
+import com.google.refine.browsing.RecordFilter;
+import com.google.refine.browsing.RowFilter;
+import com.google.refine.browsing.filters.AllRowsRecordFilter;
+import com.google.refine.browsing.filters.AnyRowRecordFilter;
+import com.google.refine.browsing.filters.ExpressionEqualRowFilter;
+import com.google.refine.browsing.util.ExpressionNominalValueGrouper;
+import com.google.refine.expr.Evaluable;
+import com.google.refine.expr.MetaParser;
+import com.google.refine.expr.ParsingException;
+import com.google.refine.model.Column;
+import com.google.refine.model.Project;
+
+public class TextListFacet extends ListFacet {
+    
+    public static class TextListFacetConfig extends ListFacetConfig {
+        @JsonProperty("selectString")
+        public boolean selectString;
+        @JsonProperty("selectNumeric")
+        public boolean selectNumeric;
+        @JsonProperty("selectDateTime")
+        public boolean selectDateTime;
+        @JsonProperty("selectBoolean")
+        public boolean selectBoolean;
+        
+        @Override
+        public String getJsonType() {
+            return "textlist";
+        }
+    }
+    
+    /*
+     * Computed results
+     */
+    protected int _stringCount;
+    protected int _numericCount;
+    protected int _datetimeCount;
+    protected int _booleanCount;
+    
+    TextListFacetConfig _config = new TextListFacetConfig();
```

Comment:
Here is a sign that inheritance is not the right tool for the job: re-declaring a `_config` attribute which was already defined in the base class is dangerous (and the cause of your NPEs). What happens in practice is that the class will have two differerent `_config` fields and methods will access one or another depending on which class they have been defined in.